plugins/buildingplan-lib.cpp

```diff
@@ -623,11 +623,6 @@ bool Planner::allocatePlannedBuilding(df::building_type type)
         return false;
     }
 
-    for (auto iter = newinst->jobs.begin(); iter != newinst->jobs.end(); iter++)
-    {
-        (*iter)->flags.bits.suspend = true;
```

Comment:
Out of curiosity, why did this move to addPlannedBuilding?

Reply:
addPlannedBuilding is an API method, but it didn't seem to enforce the supposed invariant that planned buildings are suspended. Refactoring the functionality into the method itself seemed cleaner than making it a separate API method or reimplementing the suspension in Lua.

Reply:
Ah, I see allocatePlannedBuilding calling addPlannedBuilding now (the door check was between this removed section and that call, but didn't need to be).